[CI] AsyncSearchActionIT testFinalResponseLargerMaxSize failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/n6ncap56xptdy/tests/:x-pack:plugin:async-search:internalClusterTest/org.elasticsearch.xpack.search.AsyncSearchActionIT/testFinalResponseLargerMaxSize

**Reproduction line:**
`./gradlew ':x-pack:plugin:async-search:internalClusterTest' --tests "org.elasticsearch.xpack.search.AsyncSearchActionIT.testFinalResponseLargerMaxSize" -Dtests.seed=D5AEB927BBCBD16A -Dtests.locale=zh-Hant-TW -Dtests.timezone=Africa/Nairobi -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.search.AsyncSearchActionIT&tests.test=testFinalResponseLargerMaxSize

**Failure excerpt:**
```
java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Can't store an async search response larger than [1000] bytes. This limit can be set by changing the [search.max_async_search_response_size] setting.

  at __randomizedtesting.SeedInfo.seed([D5AEB927BBCBD16A:C5236C5A12FBB558]:0)
  at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:262)
  at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:249)
  at org.elasticsearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:76)
  at org.elasticsearch.xpack.search.AsyncSearchIntegTestCase.submitAsyncSearch(AsyncSearchIntegTestCase.java:150)
  at org.elasticsearch.xpack.search.AsyncSearchActionIT.testFinalResponseLargerMaxSize(AsyncSearchActionIT.java:481)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.IllegalArgumentException: Can't store an async search response larger than [1000] bytes. This limit can be set by changing the [search.max_async_search_response_size] setting.

    at org.elasticsearch.xpack.core.async.AsyncTaskIndexService$ReleasableBytesStreamOutputWithLimit.ensureCapacity(AsyncTaskIndexService.java:652)
    at org.elasticsearch.common.io.stream.BytesStreamOutput.writeByte(BytesStreamOutput.java:67)
    at org.elasticsearch.common.io.stream.StreamOutput.write(StreamOutput.java:482)
    at java.io.FilterOutputStream.write(FilterOutputStream.java:87)
    at java.io.FilterOutputStream.write(FilterOutputStream.java:137)
    at java.util.Base64$EncOutputStream.writeb4(Base64.java:927)
    at java.util.Base64$EncOutputStream.write(Base64.java:950)
    at java.util.Base64$EncOutputStream.write(Base64.java:912)
    at java.io.FilterOutputStream.write(FilterOutputStream.java:87)
    at java.io.FilterOutputStream.write(FilterOutputStream.java:137)
    at java.util.zip.DeflaterOutputStream.deflate(DeflaterOutputStream.java:252)
    at java.util.zip.DeflaterOutputStream.finish(DeflaterOutputStream.java:225)
    at java.util.zip.DeflaterOutputStream.close(DeflaterOutputStream.java:237)
    at org.elasticsearch.common.compress.DeflateCompressor$2.close(DeflateCompressor.java:193)
    at java.io.FilterOutputStream.close(FilterOutputStream.java:191)
    at org.elasticsearch.common.io.stream.OutputStreamStreamOutput.close(OutputStreamStreamOutput.java:39)
    at org.elasticsearch.xpack.core.async.AsyncTaskIndexService.writeResponse(AsyncTaskIndexService.java:601)
    at org.elasticsearch.xpack.core.async.AsyncTaskIndexService.lambda$createResponse$2(AsyncTaskIndexService.java:260)
    at org.elasticsearch.common.xcontent.XContentBuilder.lambda$directFieldAsBase64$24(XContentBuilder.java:1084)
    at org.elasticsearch.common.xcontent.json.JsonXContentGenerator.writeDirectField(JsonXContentGenerator.java:438)
    at org.elasticsearch.common.xcontent.XContentBuilder.directFieldAsBase64(XContentBuilder.java:1074)
    at org.elasticsearch.xpack.core.async.AsyncTaskIndexService.createResponse(AsyncTaskIndexService.java:260)
    at org.elasticsearch.xpack.search.TransportSubmitAsyncSearchAction$1.onResponse(TransportSubmitAsyncSearchAction.java:87)
    at org.elasticsearch.xpack.search.TransportSubmitAsyncSearchAction$1.onResponse(TransportSubmitAsyncSearchAction.java:76)
    at org.elasticsearch.xpack.search.AsyncSearchTask.internalAddCompletionListener(AsyncSearchTask.java:255)
    at org.elasticsearch.xpack.search.AsyncSearchTask.lambda$addCompletionListener$1(AsyncSearchTask.java:191)
    at org.elasticsearch.xpack.search.AsyncSearchTask.addInitListener(AsyncSearchTask.java:277)
    at org.elasticsearch.xpack.search.AsyncSearchTask.addCompletionListener(AsyncSearchTask.java:182)
    at org.elasticsearch.xpack.search.TransportSubmitAsyncSearchAction.doExecute(TransportSubmitAsyncSearchAction.java:75)
    at org.elasticsearch.xpack.search.TransportSubmitAsyncSearchAction.doExecute(TransportSubmitAsyncSearchAction.java:44)
    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:77)
    at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:53)
    at org.elasticsearch.tasks.TaskManager.registerAndExecute(TaskManager.java:163)
    at org.elasticsearch.client.node.NodeClient.executeLocally(NodeClient.java:100)
    at org.elasticsearch.client.node.NodeClient.doExecute(NodeClient.java:80)
    at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:375)
    at org.elasticsearch.client.FilterClient.doExecute(FilterClient.java:54)
    at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:375)
    at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:364)
    at org.elasticsearch.xpack.search.AsyncSearchIntegTestCase.submitAsyncSearch(AsyncSearchIntegTestCase.java:150)
    at org.elasticsearch.xpack.search.AsyncSearchActionIT.testFinalResponseLargerMaxSize(AsyncSearchActionIT.java:481)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:568)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
    at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java
+++ b/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java
@@ -23,10 +23,8 @@ import org.elasticsearch.search.aggregations.metrics.InternalMin;
 import org.elasticsearch.search.builder.SearchSourceBuilder;
 import org.elasticsearch.test.ESIntegTestCase.SuiteScopeTestCase;
 import org.elasticsearch.xpack.core.XPackPlugin;
-import org.elasticsearch.xpack.core.async.GetAsyncResultRequest;
 import org.elasticsearch.xpack.core.search.action.AsyncSearchResponse;
 import org.elasticsearch.xpack.core.search.action.AsyncStatusResponse;
-import org.elasticsearch.xpack.core.search.action.GetAsyncSearchAction;
 import org.elasticsearch.xpack.core.search.action.SubmitAsyncSearchRequest;
 
 import java.util.ArrayList;
@@ -464,43 +462,31 @@ public class AsyncSearchActionIT extends AsyncSearchIntegTestCase {
         ensureTaskNotRunning(response.getId());
     }
 
-    public void testFinalResponseLargerMaxSize() throws Exception {
+    public void testMaxResponseSize() throws Exception {
         SearchSourceBuilder source = new SearchSourceBuilder()
             .query(new MatchAllQueryBuilder())
             .aggregation(AggregationBuilders.terms("terms").field("terms.keyword").size(numKeywords));
 
-        int limit = 1000; // should be enough to store initial response, but not enough for final response
+        final SubmitAsyncSearchRequest request = new SubmitAsyncSearchRequest(source, indexName)
+            .setWaitForCompletionTimeout(TimeValue.timeValueSeconds(10))
+            .setKeepOnCompletion(true);
+
+        int limit = 1000; // is not big enough to store the response
         ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
         updateSettingsRequest.transientSettings(Settings.builder().put("search.max_async_search_response_size", limit + "b"));
         assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
 
-        final SubmitAsyncSearchRequest request = new SubmitAsyncSearchRequest(source, indexName);
-        request.setWaitForCompletionTimeout(TimeValue.timeValueMillis(0));
+        ExecutionException e = expectThrows(ExecutionException.class,
+            () -> submitAsyncSearch(request));
+        assertNotNull(e.getCause());
 
-        // initial response – ok
-        final AsyncSearchResponse initialResponse = submitAsyncSearch(request);
-        assertTrue(initialResponse.isRunning());
-        assertNull(initialResponse.getFailure());
-
-        // final response – with failure; test that stored async search response is updated with this failure
-        assertBusy(() -> {
-            final AsyncSearchResponse finalResponse = client().execute(GetAsyncSearchAction.INSTANCE,
-                new GetAsyncResultRequest(initialResponse.getId())
-                    .setWaitForCompletionTimeout(TimeValue.timeValueMillis(300))).get();
-            assertNotNull(finalResponse.getFailure());
-            assertFalse(finalResponse.isRunning());
-            if (finalResponse.getFailure() != null) {
-                assertEquals("Can't store an async search response larger than [" + limit + "] bytes. " +
-                        "This limit can be set by changing the [" + MAX_ASYNC_SEARCH_RESPONSE_SIZE_SETTING.getKey() + "] setting.",
-                    finalResponse.getFailure().getMessage());
-            }
-        });
+        assertThat(e.getCause(), instanceOf(IllegalArgumentException.class));
+        assertEquals("Can't store an async search response larger than [" + limit + "] bytes. " +
+                "This limit can be set by changing the [" + MAX_ASYNC_SEARCH_RESPONSE_SIZE_SETTING.getKey() + "] setting.",
+            e.getCause().getMessage());
 
         updateSettingsRequest = new ClusterUpdateSettingsRequest();
         updateSettingsRequest.transientSettings(Settings.builder().put("search.max_async_search_response_size", (String) null));
         assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
-
-        deleteAsyncSearch(initialResponse.getId());
-        ensureTaskRemoval(initialResponse.getId());
     }
 }

